### ISSUE TITLE
test-client.c: why ssl_cert_filepath and not client_ssl_cert_filepath ?

### ISSUE THREAD

**Issue by dvergeylen** [warmcat/libwebsockets#1022]
Hello,

I ran accross the test-client example and was wondering why should we set up `ssl_cert_filepath` instead of `client_ssl_cert_filepath` when using SSL if the purpose is to provide *valid SSL certificate to the server* (according to comment).

Same reasoning goes to `ssl_private_key_filepath` and `client_ssl_private_key_filepath`.

https://github.com/warmcat/libwebsockets/blob/1ee5d7220d5a15dd4f961680d13381eb987f42ae/test-server/test-client.c#L576-L584

In libwebsocket.h, the comment of `client_ssl_cert_filepath` is the same as the one for `ssl_cert_filepath` which confuses a bit.
https://github.com/warmcat/libwebsockets/blob/1ee5d7220d5a15dd4f961680d13381eb987f42ae/lib/libwebsockets.h#L2226-L2229

Shouldn't they be pointer to files for client authentication? :thinking:

**Comment by lws-team** [comment]
Historically (7 years ago) lws only supported server and one vhost.  Then it supported an either-or client mode, and vhosts were split out owning the ssl contexts and listen sockets (which is why the same info struct is used for both).  Finally, nowadays any number of mixed ssl / non-ssl server and client connections can be mixed freely in any number of vhosts.

Users created their code consistent with each step as that evolved, so lws has a lot of default backwards compatibility stuff like by default creating the context creates one default vhost at the same time, as old code expects.

At the other end of the sausage machine producing new and shiny stuff, in lwsws case, the vhost is set up in one step from a JSON description, since lwsws also supports protocol plugins that want to use SSL client connections, it became necessary to be able to configure both vhost server ssl ctx and vhost client ssl ctx in one step.  So the `.client_ssl_...` members were added, for backwards compatibility the original members are used when setting up client if the .client_ssl ones are NULL.

Test-client still uses the old non-client ssl ones, it's supported so no problem but you're right, the test apps are supposed to show the most preferred way to do things.  I pushed a patch on master clarifying the .client_ssl... purpose, noting on the .ssl_... members about their backwards-compatibility use but that the .client_ssl... ones should be used for client, and update the test client to use the preferred .client_ssl... members.
